polyfill/intersectionobserver-polyfill.js

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2016 Google Inc. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+(function() {
+  // TODO: rootMargin are completely ignored for now
+
+  // Constructor
+  window.IntersectionObserver = function(callback, options) {
+    options = options || {};
+
+    if(!(callback instanceof Function)) {
+      throw('callback needs to be a function');
+    }
+
+    if(options.root && !(options.root instanceof HTMLElement)) {
+      throw('Target needs to be an HTMLelement');
+    }
+
+    this._callback = callback;
+    this._root = options.root || null;
+    this._rootMargin = options.rootMargin || [0, 0, 0, 0];
+    this._thresholds = options.threshold || 0;
+    this._init();
+  };
+
+  window.IntersectionObserver.prototype = {
```

Comment:
Rather than these repeated references to global scope, it would be better to fully compose the `IntersectionObserver` object and then at the end assign it to the global scope, eg `global.IntersectionObserver = IntersectionObserver`.  This makes it easier to potentially add a UMD or otherwise change the way the functionality of the module is exported at a later date.